### PR TITLE
Add table azure_monitor_activity_log_event Closes #621

### LINKED ISSUE #621: Add table azure_monitor_activity_log_event
**Description**
I was unable to find an existing table that would allow me to query the Azure Monitor Activity Log for a given subscription. My use case is to cross-reference Activity-Log activity with other event sources.

**References**
[AZ CLI Reference for `az monitor activity log list`](https://learn.microsoft.com/en-us/cli/azure/monitor/activity-log?view=azure-cli-latest#az-monitor-activity-log-list)

### PR DESCRIPTION
**Pull request by ParthaI**
# Integration test logs
<details>
  <summary>Logs</summary>
  
```
N/A
```
</details>

# Example query results
<details>
  <summary>Results</summary>
  
```
> select
  event_name,
  event_data_id,
  id,
  correlation_id,
  level,
  resource_id,
  event_timestamp
from
  azure_monitor_activity_log_event;
+--------------+--------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
| event_name   | event_data_id                        | id                                                                                                                                                                                                                   >
+--------------+--------------------------------------+---------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------->
| EndRequest   | eb2786b3-9bb3-4b3d-874e-103e2a124f20 | /subscriptions/je8374th-f95f-4771-bbb5-hsje384658ke/resourceGroups/DefaultResourceGroup-EUS/providers/Microsoft.OperationalInsights/workspaces/DefaultWorkspace-je8374th-f95f-4771-bbb5-hsje384658ke-EUS/linkedServic>
| BeginRequest | f1ee5ccf-9301-44e8-b502-4dcb84d248e0 | /subscriptions/je8374th-f95f-4771-bbb5-hsje384658ke/providers/Microsoft.Security/datascanners/StorageDataScanner/events/f1ee5ccf-9301-44e8-b502-4dcb84d248e0/ticks/638321063547517579                                >


> select
  event_name,
  event_timestamp,
  operation_name,
  resource_id,
  resource_type,
  status
from
  azure_monitor_activity_log_event
where
  event_timestamp >= now() - interval '300 minutes';
+--------------+---------------------------+--------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------>
| event_name   | event_timestamp           | operation_name                                                     | resource_id                                                                                                                                                >
+--------------+---------------------------+--------------------------------------------------------------------+------------------------------------------------------------------------------------------------------------------------------------------------------------>
| EndRequest   | 2023-10-05T18:09:47+05:30 | Microsoft.EventGrid/register/action                                | /subscriptions/je8374th-f95f-4771-bbb5-hsje384658ke/providers/Microsoft.EventGrid                                                                          >
| EndRequest   | 2023-10-05T18:59:51+05:30 | Microsoft.OperationalInsights/workspaces/linkedServices/write      | /subscriptions/je8374th-f95f-4771-bbb5-hsje384658ke/resourceGroups/DefaultResourceGroup-EUS/providers/Microsoft.OperationalInsights/workspaces/DefaultWorks

> select
  event_name,
  operation_name,
  event_timestamp,
  http_request ->> 'ClientRequestID' as client_request_id,
  http_request ->> 'ClientIPAddress' as ClientIPAddress,
  http_request ->> 'Method' as method,
  http_request ->> 'URI' as uri
from
  azure_monitor_activity_log_event;
+--------------+---------------------------------------------------------------------------------------------------+---------------------------+-------------------+-----------------+--------+--------+
| event_name   | operation_name                                                                                    | event_timestamp           | client_request_id | clientipaddress | method | uri    |
+--------------+---------------------------------------------------------------------------------------------------+---------------------------+-------------------+-----------------+--------+--------+
| BeginRequest | Microsoft.EventGrid/register/action                                                               | 2023-10-05T18:09:16+05:30 | <null>            | <null>          | <null> | <null> |
| BeginRequest | Microsoft.EventGrid/register/action                                                               | 2023-10-05T18:07:59+05:30 | <null>            | <null>          | <null> | <null> |
| EndRequest   | microsoft.insights/actiongroups/write                                                             | 2023-10-05T16:12:36+05:30 | <null>            | <null>          | <null> | <null> |

```
</details>

### CODE REVIEW

File: azure/table_azure_monitor_activity_log_event.go [hunk 1]
```diff
@@ -0,0 +1,566 @@
...
+				Transform:   transform.FromField("CorrelationID"),
+			},
+			{
+				Name:        "level",
+				Description: "The event level. Possible values include: 'EventLevelCritical', 'EventLevelError', 'EventLevelWarning', 'EventLevelInformational', 'EventLevelVerbose'.",
+				Type:        proto.ColumnType_STRING,
+			},
+			{
+				Name:        "resource_id",
+				Description: "The resource uri that uniquely identifies the resource that caused this event.",
```
Comment (madhushreeray30) [inline azure/table_azure_monitor_activity_log_event.go:87]:
```suggestion
				Description: "The resource URI that uniquely identifies the resource that caused this event.",
```

File: azure/table_azure_monitor_activity_log_event.go [hunk 2]
```diff
@@ -0,0 +1,566 @@
...
+			},
+			{
+				Name:        "event_name",
+				Description: "The event name. This value should not be confused with OperationName. For practical purposes, OperationName might be more appealing to end users.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("EventName.Value"),
+			},
+			{
+				Name:        "category",
+				Description: "The event category",
```
Comment (madhushreeray30) [inline azure/table_azure_monitor_activity_log_event.go:117]:
```suggestion
				Description: "The event category.",
```

File: azure/table_azure_monitor_activity_log_event.go [hunk 3]
```diff
@@ -0,0 +1,566 @@
...
+			},
+			{
+				Name:        "status",
+				Description: "A string describing the status of the operation. Some typical values are: Started, In progress, Succeeded, Failed, Resolved.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Status.Value"),
+			},
+			{
+				Name:        "sub_status",
+				Description: "The event sub status. Most of the time, when included, this captures the HTTP status code of the REST call. Common values are: OK (HTTP Status Code: 200), Created (HTTP Status Code: 201), Accepted (HTTP Status Code: 202), No Content (HTTP Status Code: 204), Bad Request(HTTP Status Code: 400), Not Found (HTTP Status Code: 404), Conflict (HTTP Status Code: 409), Internal Server Error (HTTP Status Code: 500), Service Unavailable (HTTP Status Code:503), Gateway Timeout (HTTP Status Code: 504)",
```
Comment (madhushreeray30) [inline azure/table_azure_monitor_activity_log_event.go:147]:
```suggestion
				Description: "The event sub status. Most of the time, when included, this captures the HTTP status code of the REST call. Common values are: OK (HTTP Status Code: 200), Created (HTTP Status Code: 201), Accepted (HTTP Status Code: 202), No Content (HTTP Status Code: 204), Bad Request(HTTP Status Code: 400), Not Found (HTTP Status Code: 404), Conflict (HTTP Status Code: 409), Internal Server Error (HTTP Status Code: 500), Service Unavailable (HTTP Status Code:503), Gateway Timeout (HTTP Status Code: 504).",
```

File: azure/table_azure_monitor_activity_log_event.go [hunk 4]
```diff
@@ -0,0 +1,566 @@
...
+	subscriptionID := session.SubscriptionID
+
+	client := insights.NewActivityLogsClientWithBaseURI(session.ResourceManagerEndpoint, subscriptionID)
+	client.Authorizer = session.Authorizer
+
+	filter := buildActivityLogFilter(d.Quals)
+
+	result, err := client.List(ctx, filter, "")
+	if err != nil {
+		plugin.Logger(ctx).Error("azure_monitor_activity_log_event.listMonitorActivityLogEvents", "list", err)
```
Comment (madhushreeray30) [inline azure/table_azure_monitor_activity_log_event.go:211]:
```suggestion
		plugin.Logger(ctx).Error("azure_monitor_activity_log_event.listMonitorActivityLogEvents", "client_error", err)
```

File: azure/table_azure_monitor_activity_log_event.go [hunk 5]
```diff
@@ -0,0 +1,566 @@
...
+					}
+				}
+			}
+		}
+
+	}
+
+	return filter
+}
+
```
Comment (madhushreeray30) [inline azure/table_azure_monitor_activity_log_event.go:306]: If we are not keeping the hydrate function we should remove this commented code.

**Review by madhushreeray30 — COMMENTED** [review]
@ParthaI Please can you have a look at the comments, thanks!

File: azure/table_azure_monitor_activity_log_event.go [hunk 6]
```diff
@@ -0,0 +1,566 @@
...
+			// Azure standard columns
+			{
+				Name:        "resource_group",
+				Description: ColumnDescriptionResourceGroup,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("ResourceGroupName"),
+			},
+		}),
+	}
+}
```
Comment (madhushreeray30) [inline azure/table_azure_monitor_activity_log_event.go:193]: We have missed out the `tenant_id` column please add it.

Comment (ParthaI) [reply azure/table_azure_monitor_activity_log_event.go:193]: The column `subscription_id` is available for this table. It has been moved to the common columns for all the tables. Do we add `tenant_id` in the tables?

Comment (madhushreeray30) [reply azure/table_azure_monitor_activity_log_event.go:193]: @ParthaI not sure if we add `tenant_id` in all tables however we do have a few tables that have `tenant_id` as a column.

Comment (ParthaI) [reply azure/table_azure_monitor_activity_log_event.go:193]: @madhushreeray30, I do not think it is required for this table.

**Review by madhushreeray30 — APPROVED** [review]

File: azure/table_azure_monitor_activity_log_event.go [hunk 7]
```diff
@@ -0,0 +1,305 @@
...
+			},
+			{
+				Name:        "description",
+				Description: "The description of the event.",
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("Description").Transform(transform.ToString),
+			},
+			{
+				Name:        "event_data_id",
+				Description: "The event data Id. This is a unique identifier for an event.",
```
Comment (madhushreeray30) [inline azure/table_azure_monitor_activity_log_event.go:70]:
```suggestion
				Description: "The event data ID. This is a unique identifier for an event.",
```

**Pull request merged**